podcasts/ThemeableCollectionView.swift

```diff
@@ -26,3 +33,32 @@ class ThemeableCollectionView: UICollectionView {
         indicatorStyle = AppTheme.indicatorStyle()
     }
 }
+
+extension ThemeableCollectionView: AutoScrollCollectionViewDelegate {
+    var timer: Timer? {
+        get {
+            return nil
+        }
+        set {
+        }
+    }
+
+    func intializeAutoScrollTimer() {
+        timer = Timer.scheduledTimer(timeInterval: 3.0, target: self, selector: #selector(scrolltoNextItem), userInfo: nil, repeats: true)
+       }
+
+    @objc func scrolltoNextItem() {
+        let nextIndex = (indexPathsForVisibleItems.last?.item ?? 0) + 1
+        let indexPath = IndexPath(item: nextIndex, section: 0)
+        if nextIndex < numberOfItems(inSection: 0) {
+            scrollToItem(at: indexPath, at: .left, animated: true)
+        } else if numberOfItems(inSection: 0) > 0 {
+            scrollToItem(at: IndexPath(item: 0, section: 0), at: .left, animated: false)
+        }
+    }
+
+    func stopAutoScrollTimer() {
+        timer?.invalidate()
```

Comment:
The `timer` is not invalidated because `timer` is always `null` (you can add a breakpoint here to check).

Reply:
updated